internal/datacoord/garbage_collector.go

```diff
@@ -232,21 +232,17 @@ func (gc *garbageCollector) scan() {
 
 	for idx, prefix := range prefixes {
 		startTs := time.Now()
-		infoKeys, modTimes, err := gc.option.cli.ListWithPrefix(ctx, prefix, true)
-		if err != nil {
-			log.Error("failed to list files with prefix",
-				zap.String("prefix", prefix),
-				zap.Error(err),
-			)
-		}
-		cost := time.Since(startTs)
+		objectPathHolderChan := gc.option.cli.ListWithPrefix(ctx, prefix, true)
 		segmentMap, filesMap := getMetaMap()
```

Comment:
After these PR.
we fetch meta map snapshot first, then list the keys on `oss`.
new file set meta after these operation cannot be seen in meta map snapshot.
Once our clear file loop execute too long, new file's modify time may be greater than the tolerance time.
Then oss file lost. 

Do runtime check of meta storage is safe.
Reference: `https://github.com/milvus-io/milvus/pull/29733/files#diff-47c43fd5ce193c789e97c4b71869b906dc56ed1ef694c4277510afaa67ff8e40`